### PR TITLE
Add support for running javascript functions

### LINKED ISSUE #12: Add support for running node functions
Given the current state of the npm buildpack, this issue deals with the ability to run a node app (but not necessarily to npm build it) at first

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes #12

### CODE REVIEW

File: detect/main.go [hunk 1]
```diff
@@ -45,21 +46,36 @@ func main() {
...
-			detect.Logger.Debug("riff Command application")
-			detect.Pass(command.BuildPlanContribution(metadata))
-			return
-		}
+	}
+
+	// Try node
+	// To be later changed to detecting whether the npm BP voted
+	// see https://github.com/projectriff/riff-buildpack/issues/14
+	if ok, err := node.DetectNode(detect, metadata); err != nil {
```
Comment (nebhale) [inline detect/main.go]: Reasonable middle-ground for now.  You're in contact with that team about getting an equivalent build plan entry created?

File: node/riff_invoker.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	err := r.layer.Contribute(func(artifact string, layer libjavabuildpack.DependencyLaunchLayer) error {
+		layer.Logger.SubsequentLine("Expanding to %s", layer.Root)
+		return libjavabuildpack.ExtractTarGz(artifact, layer.Root, 0)
+	})
+	if err != nil {
+		return err
+	}
+
+	entrypoint := filepath.Join(r.application.Root, r.functionJS)
+	command := fmt.Sprintf(`FUNCTION_URI="%s" RIFF_FUNCTION_INVOKER_PROTOCOL=http HOST=0.0.0.0 HTTP_PORT=8080 $NODE_HOME/bin/node %s/server.js`, entrypoint, r.layer.Root)
```
Comment (nebhale) [inline node/riff_invoker.go]: We're trying to get away from writing environment variables to command lines these days.  I highly recommend [writing those into `.profile.d`](https://github.com/cloudfoundry/openjdk-buildpack/blob/master/jre.go#L66) as either a single or multiple files, instead.  One of the positive byproducts is that when you SSH into the container, the environment is the same as when the process is run.  In it's current incarnation, *only* the process ever sees those values which has proven to be a negative for the Java Buildpack over the long run.

Comment (nebhale) [reply node/riff_invoker.go]: You don't need to use `$NODE_HOME/bin/node` here and I'd recommend against it.  Instead, depend on the the buildpack lifecycle to guarantee that the node is on `$PATH` and just call `node`.

File: go.sum [hunk 3]
```diff
@@ -1,25 +1,36 @@
 bou.ke/monkey v1.0.1 h1:zEMLInw9xvNakzUUPjfS4Ds6jYPqCFx3m7bRmG5NH2U=
 bou.ke/monkey v1.0.1/go.mod h1:FgHuK96Rv2Nlf+0u1OOVDpCMdsWyOFmeeketDHE7LIg=
+github.com/BurntSushi/toml v0.3.0/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
```
Comment (nebhale) [inline go.sum]: You probably also want to regularly run `go mod tidy` which will clean this dependency list to latest compatible versions.

**Comment by ericbottard** [comment]
Addressed review comments and merged as rebased/squashed 0a802ca

**Pull request closed without merging**